firebase_admin/messaging.py

```diff
@@ -418,6 +421,25 @@ def __init__(self, alert=None, badge=None, sound=None, content_available=None, c
         self.custom_data = custom_data
 
 
+class CriticalSound(object):
+    """Critical alert sound configuration that can be included in ``messaging.Aps``.
+
+    Args:
+        name: The name of a sound file in your app's main bundle or in the ``Library/Sounds``
+            folder of your app's container directory. Specify the string ``default`` to play the
+            system sound.
+        critical: Set to ``True`` to set the critical alert flag on the sound configuration
+            (optional).
+        volume: The volume for the critical alert's sound. Must be a value between 0.0 (silent)
+            and 1.0 (full volume) (optional).
+    """
+
+    def __init__(self, name, critical=None, volume=None):
+        self.name = name
```

Comment:
What's the rationale for doing input validation when the `ApsAlert` gets built instead of here when the values are first supplied?

Reply:
This is to guard against something like this:

```
cs = CriticalSound('foo', volume=0.5)
cs.volume = -1
send_message(cs)
```

Reply:
Oh, python. :-)  Sounds reasonable.